### PR TITLE
[dagster-airlift] instance tests

### PR DESCRIPTION
**Pull request by dpeng817**
Tests for AirflowInstance utility methods.

### CODE REVIEW

**Comment by dpeng817** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/23492?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#23492** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/23492?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#23487** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/23487?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#23483** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/23483?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#23481** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/23481?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#23480** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/23480?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#23444** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/23444?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @dpeng817 and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: examples/experimental/dagster-airlift/dagster_airlift/core/airflow_instance.py [hunk 1]
```diff
@@ -115,6 +116,22 @@ class TaskInfo(BaseModel):
...
+    airflow_run_id: str
+    start_date: datetime.datetime
+    end_date: datetime.datetime
+    run_type: str
+    airflow_run_config: Dict[str, Any]
+    raw_metadata: Dict[str, Any]
+
+    @property
+    def success(self) -> bool:
+        return self.state == "success"
```
Comment (schrockn) [inline examples/experimental/dagster-airlift/dagster_airlift/core/airflow_instance.py:132]: when are using `BaseModel` versus `record` versus plain ole python?

**Review by schrockn — CHANGES_REQUESTED** [review]
Adding `DagRun` seems disconnected from the added test?

**Comment by dpeng817** [comment]
@schrockn whoops. Meant to hook up DagRun here and forgot. I'll rip it out of this PR and put it in a downstream for now.

File: examples/experimental/dagster-airlift/dagster_airlift/core/airflow_instance.py (hunk 1 above)
Comment (dpeng817) [reply examples/experimental/dagster-airlift/dagster_airlift/core/airflow_instance.py]: This was mostly me just being inconsistent. Some stuff was originally basemodel for yaml parsing purposes, but I think most of these should just be record.

I changed the relevant classes to record.

**Review by schrockn — APPROVED** [review]
[yaytests](https://app.graphite.dev//settings/meme-library?org=dagster-io)

**Pull request merged**